File: p43/p43_taguchi.rb
n, s, t = 0, [], []
File.open(ARGV[0]) { |file| file.each_with_index do |line, i| 
  n = line.chomp.to_i if i == 0
  s << line.chomp.to_i if 0 < i && i <= n 
  t << line.chomp.to_i if n < i
end }
itv = []
for i in 0..(n - 1) do
  a = []
  a << t[i]
  a << s[i]
  itv << a
end
itv.sort!
ans, t = 0, 0
for i in 0..(n - 1) do
  if t < itv[i][1]
    ans += 1
    t = itv[i][0]
  end
end
p ans

File: p43/p43_taguchi.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)

func nextLine() string {
	sc.Scan()
	return sc.Text()
}

var n, _ = strconv.Atoi(nextLine())
var s = getArray()
var t = getArray()

func getArray() []int {
	a := make([]int, 0)
	for i := 0; i < n; i++ {
		j, _ := strconv.Atoi(nextLine())
		a = append(a, j)
	}
	return a
}

var itv = getItv()

func getItv() [][]int {
	itv := make([][]int, 0)
	for i := 0; i < n; i++ {
		a := make([]int, 0)
		a = append(a, t[i])
		a = append(a, s[i])
		itv = append(itv, a)
	}
	sort.Slice(itv, func(i, j int) bool { return itv[i][0] < itv[j][0] })
	return itv
}
func main() {
	ans, u := 0, 0
	for i := 0; i < n; i++ {
		if u < itv[i][1] {
			ans++
			u = itv[i][0]
		}
	}
	fmt.Println(ans)
}
